codegenerator/cli/src/hbs_templating/codegen_templates.rs

```diff
@@ -349,55 +337,120 @@ impl EntityRecordTypeTemplate {
+              RescriptRecordField::to_valid_res_name(&param.name), index
+          ));
+        }
+
+        for (index, param) in body_params.into_iter().enumerate() {
+            code.push_str(&format!(
+              "        {}: decodedEvent.body->Js.Array2.unsafe_get({})->HyperSyncClient.Decoder.toUnderlying->Utils.magic,\n",
+              RescriptRecordField::to_valid_res_name(&param.name), index
+          ));
+        }
+
+        code.push_str("      }\n    }");
+
+        code
+    }
+
+    pub fn from_config_event(config_event: &system_config::Event) -> Result<Self> {
+        let name = config_event.name.to_capitalized_options();
+        match &config_event.payload {
+            EventPayload::Params(params) => {
+                if params.is_empty() {
+                    Ok(EventTemplate {
+                        name,
+                        params: vec![],
+                        data_type: "unit".to_string(),
+                        data_schema_code: "S.literal(%raw(`null`))->S.variant(_ => ())".to_string(),
+                        topic0: config_event.topic0.to_string(),
+                        convert_hyper_sync_event_args_code:
+                            "(Utils.magic: HyperSyncClient.Decoder.decodedEvent => eventArgs)"
+                                .to_string(),
+                        decode_hyper_fuel_data_code: Self::DECODE_HYPER_FUEL_DATA_CODE.to_string(),
+                    })
+                } else {
+                    let template_params = params
+                        .iter()
+                        .map(|input| {
+                            let res_type = abi_to_rescript_type(&input.into());
+                            let js_name = input.name.to_string();
+                            EventParamTypeTemplate {
+                                res_name: RescriptRecordField::to_valid_res_name(&js_name),
+                                js_name,
+                                default_value_rescript: res_type.get_default_value_rescript(),
+                                default_value_non_rescript: res_type
+                                    .get_default_value_non_rescript(),
+                                res_type: res_type.to_string(),
+                                is_eth_address: res_type == RescriptTypeIdent::Address,
+                            }
+                        })
+                        .collect::<Vec<_>>();
+
+                    let data_type_expr = RescriptTypeExpr::Record(
+                        params
+                            .iter()
+                            .map(|p| {
+                                RescriptRecordField::new(
+                                    p.name.to_string(),
+                                    abi_to_rescript_type(&p.into()),
+                                )
+                            })
+                            .collect(),
+                    );
```

Comment:
Lets remove the above if else check and then if params are empty just make unit type expression and the rest of the logic can remain the same.

Reply:
Done